earth_enterprise/rpms/pre-install_fusion.sh

```diff
@@ -0,0 +1,167 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: requires xmllint from libxml2-utils
+
+set +x
+
+#-----------------------------------------------------------------
+BADHOSTNAMELIST=(empty linux localhost dhcp bootp)
+
+# versions and user names
+GEE="Google Earth Enterprise"
+GEES="$GEE Server"
+
+# directory locations
+BININSTALLROOTDIR="/etc/init.d"
+BASEINSTALLDIR_OPT="/opt/google"
+BASEINSTALLDIR_ETC="/etc/opt/google"
+BASEINSTALLDIR_VAR="/var/opt/google"
+
+# derived directories
+SYSTEMRC="$BASEINSTALLDIR_ETC/systemrc"
+
+# Get system info values
+NEWLINECLEANER="sed -e s:\\n::g"
+HOSTNAME="$(hostname -f | tr [A-Z] [a-z] | $NEWLINECLEANER)"
+HOSTNAME_F="$(hostname -f | $NEWLINECLEANER)"
+
+check_bad_hostname() {
+    if [ -z "$HOSTNAME" ] || [[ " ${BADHOSTNAMELIST[*]} " == *"${HOSTNAME,,} "* ]]; then
+        show_badhostname
+
+        if [ $BADHOSTNAMEOVERRIDE == true ]; then
+            echo -e "Continuing the installation process...\n"
+            # 0 = true (in bash)
+            return 0
+        else
+            echo -e "Exiting the installer.  If you wish to continue, re-run this command with the -hnf 'Hostname Override' flag.\n"
+            # 1 = false
+            return 1
+        fi
+    fi
+}
+
+show_badhostname()
+{
+    echo -e "\nYour server [$HOSTNAME] contains an invalid hostname value which typically"
+    echo -e "indicates an automatically generated hostname that might change over time."
+    echo -e "A subsequent hostname change would cause configuration issues for the "
+    echo -e "$SOFTWARE_NAME software.  Invalid values: ${BADHOSTNAMELIST[*]}."
+}
+
+check_mismatched_hostname() {
+    if [ $HOSTNAME != $HOSTNAME_F ]; then
+        show_mismatchedhostname
+
+        if [ $MISMATCHHOSTNAMEOVERRIDE == true ]; then
+            echo -e "Continuing the installation process...\n"
+            # 0 = true (in bash)
+            return 0
+        else
+            echo -e "Exiting the installer.  If you wish to continue, re-run this command with the -hnmf 'Hostname Mismatch Override' flag.\n"
+            # 1 = false
+            return 1
+        fi
+    fi
+}
+
+show_mismatchedhostname()
+{
+    echo -e "\nThe hostname of this machine does not match the fully-qualified hostname."
+    echo -e "$SOFTWARE_NAME requires that they match for local publishing to function properly."
+}
+
+#-----------------------------------------------------------------
+
+# config values
+ASSET_ROOT="/gevol/assets"
+
+#-----------------------------------------------------------------
+# Main Functions
+#-----------------------------------------------------------------
+main_preinstall()
+{
+	# check to see if GE Fusion processes are running
+	if ! check_fusion_processes_running; then
```

Comment:
We don't need a function to check if an installed service is running, and a comment won't stop a running service.

This whole `if` section should be replaced by something like:

```BASH
service gefusion stop
```

Reply:
ok